### DISCUSSION TITLE
Adding custom query with GraphQL

### DISCUSSION THREAD

**Discussion by avxkim** [strapi/strapi#6144] [Legacy Topics]
There's an example, on how to add count query: https://strapi.io/documentation/3.0.0-beta.x/guides/count-graphql.html#setup-the-application

But can i add that count query inside, for example restaurants (get all the restaurant endpoint that created by default)?

**Comment by robertpiosik** [comment]
@webcoderkz do you mean, inside controllers? If so, you can follow instructions available [here](https://strapi.io/documentation/3.0.0-beta.x/plugins/graphql.html#customise-the-graphql-schema)

**Comment by avxkim** [comment]
@robertpiosik i've read this already, but it doesn't answer my question on how to override existing query, like get all restaurants? I want to add count number into restaurants query (which created by default)

**Reply by kierangilliam** [reply]
Maybe something like this?

`api/restaurants/config/schema.graphql.js`

```
module.exports = {
    resolver: {
        Mutation: {
            find: {
                resolverOf: 'application::restaurants.restaurants.find',
                policies: [
                    'plugins::users-permissions.isAuthenticated',
                ],
                resolver: async(_, __, { context }) => {
                    const customResponse = ...
                    return customResponse
                    /** Or if you have the find controller setup for what you want to do..
                    context.params = context.request.body
                    return await strapi.api.restaurants.controllers.restaurants.find(context)
                    */
                },
            },
        },
    },
};
```
